koap/src/commonMain/kotlin/multiplatform/ObjectHasher.kt

```diff
@@ -0,0 +1,10 @@
+package com.juul.koap.multiplatform
+
+expect object ObjectHasherFactory {
+    fun createObjectHasher(): ObjectHasher
+}
```

Comment:
We have a singleton here that just creates the object on every invocation of `createObjectHasher`.
We should either utilize the singleton to either lazily cache or just set outright the object hasher to provide.
Otherwise we should just instantiate the object as needed and forgo the singleton.

Reply:
**UPDATE**: After thinking about it some more, updated my original comment.

Reply:
For `expect` it cannot be used on `override` and it cannot be `private`. I have removed this class in favor of only a single `expect` function. Not sold on my naming though.